gallery_dl/extractor/lightroom.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018-2022 Mike Fährmann
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extract images from https://lightroom.adobe.com/"""
+
+from .common import Extractor, Message
+from .. import text
+
+import json
+
+
+class LightroomGalleryExtractor(Extractor):
+    """Extractor for an image gallery on lightroom.adobe.com"""
+
+    category = "lightroom"
+    subcategory = "gallery"
+    directory_fmt = ("{category}", "{user}", "{title}")
+    filename_fmt = "{num:>04}_{id}.{extension}"
+    archive_fmt = "{id}"
+    pattern = r"(?:https?://)?lightroom\.adobe\.com/shares/([0-9a-f]+)"
```

Comment:
Wouldn't hurt having `(?:www\.)?` just in case.

Reply:
The `www` subdomain doesn't exist, so such an URL wouldn't work anyway.